### PR TITLE
fix(luyd): fix new pipeline impala in Lunarlander and Atari env

### PR DESCRIPTION
**Pull request by AltmanD**
Fix new pipeline impala in lunarlander env.

### CODE REVIEW

File: ding/data/buffer/deque_buffer.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ def __init__(self, size: int) -> None:
         super().__init__(size=size)
         self.storage = deque(maxlen=size)
         self.indices = BufferIndex(maxlen=size)
+        self.sliced = sliced
```
Comment (PaParaZz1) [inline ding/data/buffer/deque_buffer.py:67]: add the new argument comment

File: ding/data/buffer/deque_buffer.py [hunk 2]
```diff
@@ -348,8 +352,17 @@ def filter_by_unroll_len():
             seq_data = sampled_data[group]
             # Filter records by unroll_len
             if unroll_len:
-                start_indice = random.choice(range(max(1, len(seq_data) - unroll_len)))
-                seq_data = seq_data[start_indice:start_indice + unroll_len]
+                if sliced:
+                    start_indice = random.choice(range(max(1, len(seq_data))))
```
Comment (PaParaZz1) [inline ding/data/buffer/deque_buffer.py:356]: add some comments to explain reasons here

File: ding/data/buffer/deque_buffer.py [hunk 3]
```diff
@@ -348,8 +352,17 @@ def filter_by_unroll_len():
...
+                    start_indice = start_indice // unroll_len
+                    if start_indice == (len(seq_data) - 1) // unroll_len:
+                        seq_data = seq_data[-unroll_len:]
+                    else:
+                        seq_data = seq_data[start_indice * unroll_len:start_indice * unroll_len + unroll_len]
+                else:
+                    start_indice = random.choice(range(max(1, len(seq_data) - unroll_len)))
+                    seq_data = seq_data[start_indice:start_indice + unroll_len]
+
```
Comment (PaParaZz1) [inline ding/data/buffer/deque_buffer.py:365]: add new unittest for this new argument

File: ding/example/impala.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        )
+        policy = IMPALAPolicy(cfg.policy, model=model)
+
+        task.use(interaction_evaluator(cfg, policy.eval_mode, evaluator_env))
+        task.use(StepCollector(cfg, policy.collect_mode, collector_env, random_collect_size=1024))
+        task.use(data_pusher(cfg, buffer_, group_by_env=True))
+        task.use(OffPolicyLearner(cfg, policy.learn_mode, buffer_))
+        task.use(online_logger(train_show_freq=300))
+        task.use(CkptSaver(policy, cfg.exp_name, train_freq=10000))
+        task.use(termination_checker(max_env_step=15e5))
```
Comment (PaParaZz1) [inline ding/example/impala.py:43]: modify this value

File: ding/example/impala.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    cfg = compile_config(main_config, create_cfg=create_config, auto=True)
+    ding_init(cfg)
+    with task.start(async_mode=False, ctx=OnlineRLContext()):
+        collector_env = BaseEnvManagerV2(
+            env_fn=[lambda: LunarLanderEnv(cfg.env) for _ in range(cfg.env.collector_env_num)], cfg=cfg.env.manager
+        )
+        evaluator_env = BaseEnvManagerV2(
+            env_fn=[lambda: LunarLanderEnv(cfg.env) for _ in range(cfg.env.evaluator_env_num)], cfg=cfg.env.manager
+        )
+        cfg.seed = 133
```
Comment (PaParaZz1) [inline ding/example/impala.py:28]: why modify seed here

File: ding/example/impala.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+from dizoo.box2d.lunarlander.config.lunarlander_impala_config import main_config, create_config
+from dizoo.box2d.lunarlander.envs import LunarLanderEnv
+
+
+def main():
+    logging.getLogger().setLevel(logging.INFO)
+    cfg = compile_config(main_config, create_cfg=create_config, auto=True)
+    ding_init(cfg)
+    with task.start(async_mode=False, ctx=OnlineRLContext()):
+        collector_env = BaseEnvManagerV2(
```
Comment (PaParaZz1) [inline ding/example/impala.py:22]: why not subprocess

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

**Comment by AltmanD** [comment]
### Reward curve in Lunarlander, Pong, Spaceinvaders
#### Lunarlander:
image
#### Pong:
image
#### Spaceinvaders:
image

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**